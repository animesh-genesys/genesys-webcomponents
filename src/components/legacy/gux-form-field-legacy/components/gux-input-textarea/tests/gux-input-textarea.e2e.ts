import {
  newSparkE2EPage,
  a11yCheck
} from '../../../../../../../tests/e2eTestUtils';

const axeExclusions = [
  {
    issueId: 'label',
    target: 'textarea',
    exclusionReason:
      'gux-input-textarea is used within the gux-form-field component which provides a label'
  }
];

describe('gux-input-textarea', () => {
  it('renders', async () => {
    const html = `
      <gux-input-textarea lang="en">
        <textarea slot="input" type="test"></textarea>
      </gux-input-textarea>
    `;
    const page = await newSparkE2EPage({ html });
    const element = await page.find('gux-input-textarea');

    expect(element).toHaveAttribute('hydrated');
  });

  describe('#render', () => {
    [
      '<gux-input-textarea><textarea type="text" slot="input"/></textarea></gux-input-textarea>',
      '<gux-input-textarea resize="none"><textarea type="text" slot="input"/></textarea></gux-input-textarea>',
      '<gux-input-textarea resize="manual"><textarea type="text" slot="input"/></textarea></gux-input-textarea>',
      '<gux-input-textarea resize="auto"><textarea type="text" slot="input"/></textarea></gux-input-textarea>',
      '<gux-input-textarea resize="auto"><textarea type="text" slot="input"/>Test</textarea></gux-input-textarea>'
    ].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newSparkE2EPage({ html });
        const element = await page.find('gux-input-textarea');
        await a11yCheck(page, axeExclusions);

        expect(element.innerHTML).toMatchSnapshot();
      });
    });
  });
});
