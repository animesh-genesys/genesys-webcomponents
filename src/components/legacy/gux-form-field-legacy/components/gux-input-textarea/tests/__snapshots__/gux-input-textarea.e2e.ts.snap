// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-input-textarea #render should render component as expected (1) 1`] = `"<!----><div class=\\"gux-resize-none\\"><textarea type=\\"text\\" slot=\\"input\\"></textarea></div>"`;

exports[`gux-input-textarea #render should render component as expected (2) 1`] = `"<!----><div class=\\"gux-resize-none\\"><textarea type=\\"text\\" slot=\\"input\\"></textarea></div>"`;

exports[`gux-input-textarea #render should render component as expected (3) 1`] = `"<!----><div class=\\"gux-resize-manual\\"><textarea type=\\"text\\" slot=\\"input\\"></textarea></div>"`;

exports[`gux-input-textarea #render should render component as expected (4) 1`] = `"<!----><div class=\\"gux-resize-auto\\" data-replicated-value><textarea type=\\"text\\" slot=\\"input\\"></textarea></div>"`;

exports[`gux-input-textarea #render should render component as expected (5) 1`] = `"<!----><div class=\\"gux-resize-auto\\" data-replicated-value=\\"Test\\"><textarea type=\\"text\\" slot=\\"input\\">Test</textarea></div>"`;
